WebApp/src/test/java/ev/projects/webapp/controllers/CaseControllerTest.java

```diff
@@ -29,11 +41,45 @@ public void start() {
     }
 
     @Test
-    public void getCases() {
+    public void getCasesTest() throws NoSuchFieldException, IllegalAccessException {
+        List<Case> cases = MockData.generateCases();
+        addCases(cases);
         String url = baseUrl + "/api/cases/";
         TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username, password);
-        List<Case> response = restTemplate.getForObject(url, String.class);
-        System.out.println(response);
+        ResponseEntity<List<Case>> casesResponse = restTemplate.exchange(url, HttpMethod.GET,
+                null, new ParameterizedTypeReference<List<Case>>() {});
+        List<Case> responseCases = casesResponse.getBody();
+        assertNotNull(responseCases);
+        assertEquals(2, responseCases.size());
+        Case aCase = responseCases.get(1);
+        assertNotNull(aCase);
+        assertEquals(aCase.getID(), 2);
+        assertEquals(aCase.getTitle(), "Math");
+    }
+
+    @ParameterizedTest
+    @CsvSource({"1,English,Smith", "2,Math,Mike"})
+    public void getCaseTest(long ID, String title, String creatorUser) throws NoSuchFieldException, IllegalAccessException {
+        Case aCase = new Case(ID, title, new Date(), creatorUser);
+        addCases(List.of(aCase));
+        String url = baseUrl + "/api/cases/" + ID + "/";
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username, password);
+        ResponseEntity<Optional<Case>> caseResponse = restTemplate.exchange(url, HttpMethod.GET,
+                null, new ParameterizedTypeReference<Optional<Case>>() {});
+        Optional<Case> responseCase = caseResponse.getBody();
+        assertNotNull(responseCase);
+        assertTrue(responseCase.isPresent());
+        Case rCase = responseCase.get();
+        assertEquals(aCase, rCase);
+        assertEquals(ID, rCase.getID());
+        assertEquals(title, rCase.getTitle());
+        assertEquals(creatorUser, rCase.getCreatorUser());
+    }
+
+    private void addCases(List<Case> cases) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
This is a good utility method for @Before every test is run.
However a good practice is also to have the @After method to clear up the data generated by test.
Try implementing the @After method as well.

Reply:
I have done something like that in db5b6e4. Does that solve this?

Reply:
Yes, cleanUp() methods is what I expected. Perfect.